include/seastar/core/chunked_fifo.hh

```diff
@@ -210,6 +211,12 @@ private:
 
 };
 
+template <typename T, size_t items_per_chunk>
+template <typename U>
+inline
+chunked_fifo<T, items_per_chunk>::basic_iterator<U>::basic_iterator() noexcept : _chunk(nullptr), _item_index(0) {
```

Comment:
nit: can this be a delegating constructor and pass in a `nullptr` to the explicit constructor that takes a chunk?